### PR TITLE
filter: clear route cache in language filter

### PR DESCRIPTION
**Pull request by realtimetodie**
Adds a new option to the contrib http language filter to allow users to clear the route cache when the x-language header was added.

Related

- https://github.com/envoyproxy/envoy/issues/27840

### CODE REVIEW

File: api/contrib/envoy/extensions/filters/http/language/v3alpha/language.proto
Comment (ravenblackx) [inline api/contrib/envoy/extensions/filters/http/language/v3alpha/language.proto:37]: Probably worth elaborating on this comment since it generates user-facing documentation, e.g.
```
If the x-language header is altered, clear the route cache for the current request.
This should be set if the route configuration may depend on the x-language header.
Otherwise it should be unset to avoid the performance cost of route recalculation.
```

**Comment by ravenblackx** [comment]
Assigning to mattklein123@ with a question - what do we do about [CODEOWNERS for this contrib path](https://sourcegraph.com/github.com/envoyproxy/envoy/-/blob/CODEOWNERS?L361) just having the same user twice, also being a user that isn't available as a review assignee?

**Comment by ravenblackx** [comment]
Oh, never mind that question, it's answered by the next PR I was triaging. :)

**Comment by ravenblackx** [comment]
Yeah, I'm not thinking it needs a maintainer sponsor, but it does still need a maintainer to review it and merge it - the question isn't really specific to this particular folder, rather, it's supposed to be reviewed first, then /assign-from `@envoyproxy/envoy-maintainers` to get merged. But with only one owner and that being the person writing the code, I don't know who to assign as the first pass reviewer.

I guess maybe...
/assign-from @envoyproxy/first-pass-reviewers

**Comment by ravenblackx** [comment]
Looks like that should work. Once you have approval from a first-pass reviewer, you should be able to do `/assign-from @envoyproxy/envoy-maintainers` to summon someone to merge it.

**Comment by ravenblackx** [comment]
Though actually since I'm here anyway and already looked at it all, I can just do all the steps this time. And I figured out what to do assign-wise next time. :)

**Review by ravenblackx — APPROVED** [review]

**Comment by tyxia** [comment]
(receiving email from first-pass-reviewer..).We do this in ext_proc/ ext_authz filter already. Also, we are currently cleaning up clear related logic in ext_proc as clear_route_cache should not be triggered on response path.

 I can take a pass on this later.

**Comment by tyxia** [comment]
> (receiving email from first-pass-reviewer..).We do this in ext_proc/ ext_authz filter already. Also, we are currently cleaning up clear related logic in ext_proc as clear_route_cache should not be triggered on response path.
> 
> I can take a pass on this later.

Oh just noticed this is already approved. looks like we are commenting at the same time :)

**Comment by ravenblackx** [comment]
It would be nice to also have tests, but as far as I can tell that's not required for contrib (it was landed with no tests before, so...)

**Pull request merged**
